### ISSUE TITLE
loss.backward()??

### ISSUE THREAD

**Issue by a897456** [lucidrains/naturalspeech2-pytorch#38]
In Usage:
`loss = diffusion(raw_audio)`
`loss.backward()`
Thank you for your work, very nice! And I'm sorry, as a newbie, I have to ask two stupid questions:

1. Where does this backward() go? I didn't find a follow-up to it, so I had a second question
2. naturalspeech2, I don't know if it's a model or a method, so I don't know how to train it, or if I need to train it

**Comment by a897456** [comment]
![image](https://github.com/lucidrains/naturalspeech2-pytorch/assets/28752526/3790e4ac-1ad8-4f1b-b513-e5c06559d9cf)
![image](https://github.com/lucidrains/naturalspeech2-pytorch/assets/28752526/ae7780f8-7321-4318-893e-82b5d7a4c1d2)

Hi,@lucidrains 
I trained it using Part 3 of Usage, which will take 100k steps, and 1k steps per epoch, so complete the training will use100 epochs. It should be able to generate 100 `.flac` files and 100 `.pt` files. At present, I have listened to the 51st generated `.flac` file and felt that it was white noise. What's going on, please?

**Comment by a897456** [comment]
Hi @lucidrains 
Logically, when epoch=50, I should produce an audio file that doesn't sound like white noise, right?
but, so far, the output of two files sound like white noise, do you know how to solve it? Please,THS

**Comment by a897456** [comment]
https://github.com/lucidrains/naturalspeech2-pytorch/blob/659bec7f7543e7747e809e950cc2f84242fbeec7/naturalspeech2_pytorch/naturalspeech2_pytorch.py#L1700-L1701

https://github.com/lucidrains/naturalspeech2-pytorch/blob/659bec7f7543e7747e809e950cc2f84242fbeec7/naturalspeech2_pytorch/naturalspeech2_pytorch.py#L1874-L1879

HI @lucidrains 
Does this mean that only two sets of batch are involved in the loss calculation at each step?
